src/scc_hypervisor_collector/api/uploader.py

```diff
@@ -5,7 +5,59 @@
 collected from the specified backends to the SCC using the provided
 credentials.
 """
+import json
+import logging
+import gzip
+from typing import Dict
+from importlib_metadata import version
+import requests
+from requests.exceptions import RequestException
+from .configuration import SccCredsConfig
 
 
 class SCCUploader:
     """SCC Uploader for scc-hypervisor-collector."""
+
+    def __init__(self, scccredsconfig: SccCredsConfig):
+        """Initialiser for SCCUploader"""
+        self._log = logging.getLogger(__name__)
+
+        # save the parameters
+        self._scccredsconfig: SccCredsConfig = scccredsconfig
+        self.username = self._scccredsconfig.username
+        self.password = self._scccredsconfig.password
```

Comment:
Do we need to define these two attributes?

Since the only place they are used is in the `self.auth` assignment below, they are needed by the class implementation from what I can see. And defining them like this makes them assignable attributes of the class instance, but since they are not used outside of the `__init__()` method, any such assignments would have no effect.

So probably drop then, or else convert then to `@property` methods if you want to expose them for readonly retrieval...

And if we just want to expose them